db_api.rkt

```diff
@@ -129,17 +132,13 @@
  (define/ctor (view-from-struct v-struct)
    v-struct)
 
- ; Needed by join method to the underlying struct from the other view capability
- (define/op (get-struct)
-   (view-impl-v this))
-
  ; Methods that return new view capabilities
  (define/op (where cond)
    (view-from-struct (where-impl (view-impl-v this) cond)))
  (define/op (select cols)
    (view-from-struct (select-impl (view-impl-v this) cols)))
  (define/op (join v2 jcond)
-   (view-from-struct (join-impl (view-impl-v this) (get-struct v2) jcond)))
+   (view-from-struct (join-impl (view-impl-v this) (view-impl-v v2) jcond)))
```

Comment:
I removed `get-struct` from the view interface since it is only required in the implementation of `view-impl`. Here we have the capability's accessors in scope and can just access the implementation's fields directly.